### ISSUE TITLE
PolylinePipeline.generateCartesianArc fails on two equal positions

### ISSUE THREAD

**Issue by nkbt** [AnalyticalGraphicsInc/cesium#2296]
generateCartesianArc removes duplicate coordinates even if we have only two

Exception:
```
An error occurred while rendering. Rendering has stopped.
DeveloperError: right is required
Error
    at new DeveloperError (http://localhost:3000/assets/cesium/Workers/createGeometry.js:146:19)
    at Function.Cartesian3.subtract (http://localhost:3000/assets/cesium/Workers/createGeometry.js:1528:19)
    at Function.PolylineGeometry.createGeometry (http://localhost:3000/assets/cesium/Workers/createPolylineGeometry.js:15668:28)
    at createPolylineGeometry (http://localhost:3000/assets/cesium/Workers/createPolylineGeometry.js:15823:33)
    at createGeometry (http://localhost:3000/assets/cesium/Workers/createGeometry.js:19170:30)
    at DedicatedWorkerGlobalScope.createTaskProcessorWorker (http://localhost:3000/assets/cesium/Workers/createGeometry.js:19065:42)
```

Code I used to make this happen
```
var primitive = new Cesium.Primitive({
  geometryInstances: new Cesium.GeometryInstance({
    geometry: new Cesium.PolylineGeometry({
      positions: Cesium.Cartesian3.fromDegreesArray([104, -4, 104, -4]),
      width: this.settings.atkWidth,
      vertexFormat: Cesium.PolylineColorAppearance.VERTEX_FORMAT
    }),
    attributes: {
      color: Cesium.ColorGeometryInstanceAttribute.fromColor(new Cesium.Color(0.9, 1.0, 0.1, 1))
    }
  }),
  appearance: new Cesium.PolylineColorAppearance()
});

widget.scene.primitives.add(primitive);
```

I traced the problem and it appears to be here:

cesium/Workers/createPolylineGeometry.js:15627
```
positions = PolylinePipeline.generateCartesianArc({
    positions: positions,
    granularity: granularity,
    ellipsoid: ellipsoid,
    height: heights
});
```

it returns `[{x: -1539277.3670999212, y: 6173704.315387618, z: -441945.1052480023, __proto__: Cartesian3}]`

So we end up only with one single Cartesian3 position and further code fails here:

cesium/Workers/createPolylineGeometry.js:15668
```
var positionsLength = positions.length;
for (j = 0; j < positionsLength; ++j) {
    if (j === 0) {
        position = scratchCartesian3;
        Cartesian3.subtract(positions[0], positions[1], position); // <--- FAILS here
        Cartesian3.add(positions[0], position, position);
    } else {
        position = positions[j - 1];
    }

```

**Comment by zhhwGIS** [comment]
i have the same problem! And how do you solve it?

**Comment by nkbt** [comment]
@zhhwGIS I made my own build from master branch. That worked for me. I reckon that might be fixed in 1.4

**Comment by mramato** [comment]
Thanks for the report. With the recent deduplication fixes and new policy for Primtivies that contain an invalid number of unique points (they no longer try to draw rather than throw exceptions); this no longer appears to crash.  If either of you are still having problems, let us know.
